File: Cousera DSA/Algorithmic ToolBox/Week2/Assignment2Ex/02_fib_last_digit_problem.cpp
#include <iostream>
#include <gmp.h>

using namespace std;

long long fibEfficient(int n){

    long long arr[n];
    arr[0] = 1;
    arr[1] = 1;
    
    for (int i = 2; i < n; i++)
    {
        arr[i] = arr[i-1] + arr[i-2];
    }
    
    return arr[n-1];
    
}

short int lastBitFib(int number){
    return fibEfficient(number) % 10;
}

int main(int argc, char const *argv[])
{
    int n;
    cin >> n;

    cout << lastBitFib(n) << endl;
    return 0;
}

File: Cousera DSA/Algorithmic ToolBox/Week2/Assignment2Ex/02_fib_last_digit_problem.py
def fibEfficient(n):
    fib_arr = []
    fib_arr.append(1)
    fib_arr.append(1)
    for i in range(2,n):
        fib_arr.append(fib_arr[i-1]+fib_arr[i-2])
    
    return fib_arr[n-1]

def lastBitFib(n):
    res = fibEfficient(n)
    return res % 10

n = int(input())
print(lastBitFib(n))

